examples/can.rs

```diff
@@ -46,36 +48,47 @@ fn main() -> ! {
         .into_af9_push_pull(&mut gpioa.moder, &mut gpioa.otyper, &mut gpioa.afrh);
 
     // Initialize the CAN peripheral
-    let can = Can::new(dp.CAN, rx, tx, &mut rcc.apb1);
+    let can = Can::new(dp.CAN, tx, rx, &mut rcc.apb1);
 
-    // Uncomment the following line to enable CAN interrupts
-    // can.listen(Event::Fifo0Fmp);
+    let mut can = bxcan::Can::new(can);
 
-    let (mut tx, mut rx0, _rx1) = can.split();
+    // Use loopback mode: No pins need to be assigned to peripheral.
+    // APB1 (PCLK1): 64MHz, Bit rate: 500kBit/s, Sample Point 87.5%
+    // Value was calculated with http://www.bittiming.can-wiki.info/
+    can.modify_config()
+        .set_bit_timing(0x001c_0003)
+        .set_loopback(false)
+        .set_silent(false);
+
+    let mut filters = can.modify_filters();
+
+    filters.enable_bank(0, Mask32::accept_all());
+
+    // Enable filters.
+    drop(filters);
```

Comment:
This seems confusing. Has the `Drop` trait the site effect of applying the filter?

Reply:
Ah I see, filter holds a mutual reference to can, so can can not be used later, as long as the reference is held. Is this correct? And does it still compile without this drop?